rep-0105.rst

```diff
@@ -81,9 +83,31 @@ The coordinate frame called ``base_link`` is rigidly attached to the
 mobile robot base. The ``base_link`` can be attached to the base in
 any arbitrary position or orientation; for every hardware platform
 there will be a different place on the base that provides an obvious
-point of reference. Note that REP 103 [1] specifies a preferred
+point of reference. Note that REP 103 [1]_ specifies a preferred
 orientation for frames.
 
+earth
+'''''
+
+The coordinate frame called ``earth`` is the origin of ECEF.
+
+This frame is designed to allow the interaction of multiple robots in different map frames.
+If the application only needs one map the ``earth`` coordinate frame is not expected to be present.
+In the case of running with multiple maps simultaneously the ``map`` and ``odom`` and ``base_link`` frames will need to be customized for each robot.
+If running multiple robots and bridging data between them, the transform frame_ids can remain standard on each robot if the other robots_frame_ids are rewritten.
+
+If the ``map`` frame is globally referenced the publisher from ``earth`` to ``map`` can be a static transform publisher.
+Otherwise the ``earth`` to ``map`` transform will usually need to be computed by taking the estimate of the current global position and subtracting the current estimated pose in the map to get the estimated pose of the origin of the map.
+
+In case the ``map`` frame's absolute positon is unknown at the time of startup, it can remain detached until such time that the global position estimation can be adaquately evaluated.
+This will operate in the same way that a robot can operate in the ``odom`` frame before localization in the ``map`` frame is innitialized.
+
+.. figure:: rep-0105/ECEF_ENU_Longitude_Latitude_relationships.svg
+   :width: 640px
+   :alt: Earth Centered Earth Fixed diagram
+   
+   A visualization of Earth Centered Earth Fixed with a tangential ``map`` frame.
+
```

Comment:
> With a connected tree, tf should be able to transform a point from any map into any other map. And there's the globally oriented earth coordinate frame.

That is helpful.

> For long distance planning I would expect it to use the global frame or something else which is specific to the non-euclidian planning.

Probably lat/long in most applilcations.

> When a vehicle is travelling it will need to reparent to a new map periodically, such as when you transition between UTM zones.

That is the crux of my question. It needs to be done even more frequently than when crossing UTM boundaries. Although TF uses double-precision, most sensor data are single-precision float, which need to be transformed every few kilometers to avoid excessive loss of accuracy. 

I really don't know how to do that. My best guess would involve introducing something like a new ``odom`` frame, then notifying every package caching obstacle data to perform the appropriate transform. That seems messy, as it affects multiple ROS components (move_base, etc.). 

So, how should reparenting be handled in a consistent way that works with multiple ROS packages? That is a legitimate question for the REP to resolve. I have not thought about this problem is quite a while. Maybe I am missing some obvious answer.

>  Does that clarify what you're asking?

Yes, thanks.

> (Aside: Would a node that publishes all UTM 60 zones as static transforms from earth be useful?)

Not that I can see. As long as ``geodesy`` or some similar package provides convenient conversions between the relevant coordinate types, such a node could easily be constructed if needed.